### ISSUE TITLE
No rule to make target 'opencv_python3'

### ISSUE THREAD

**Issue by bobuntu** [opencv/opencv#15280]
##### System information (version)
- OpenCV => 4.1.1
- Operating System / Platform => Ubuntu 18.04.3 LTS
- Compiler => GCC-7
- CUDA 10.0
##### Detailed description
After generating the build files using CMake, I noticed that python2 and python3 were listed among the "**Unavailable**" modules to be built. The following command fails:
```bash
cmake --build . --target opencv_python3
make: *** No rule to make target 'opencv_python3'. Stop.
```
##### Steps to reproduce
 Here is the command I used:
```bash
cmake -DBUILD_PYTHON_SUPPORT=ON \
  -DWITH_V4L=ON \
  -DBUILD_NEW_PYTHON_SUPPORT=ON \
  -DINSTALL_PYTHON_EXAMPLES=ON \
  -DOPENCV_PYTHON3_INSTALL_PATH=/home/<user>/<venv-path>/lib/python3.6/site-packages ..
```
>Note: I performed these operations under a python virtual environment

**Comment by bobuntu** [comment]
**TL;DR** Set `-DPYTHON_DEFAULT_EXECUTABLE=/path/to/bin/python` ¯\_(ツ)_/¯

I'm not sure what I did that changed, but I kept playing around with the **cmake -D tags** until I could get the cmake target `opencv_python3` generated. I introduced some (hidden) flags to the build command after reading from a few posts that the `OPENCV_FORCE_PYTHON_LIBS` could be blamed for this issue. Here is the build command:
```bash
cmake -DPYTHON_DEFAULT_EXECUTABLE=/path/to/bin/python \
-DBUILD_PYTHON_SUPPORT=ON \
-DWITH_V4L=ON \
-DBUILD_NEW_PYTHON_SUPPORT=ON \
-DINSTALL_PYTHON_EXAMPLES=ON \
-DOPENCV_PYTHON3_INSTALL_PATH=/path/to/lib/python3.6/site-packages \
-DCMAKE_C_COMPILER=/usr/bin/gcc \
-DCMAKE_CXX_COMPILER=/usr/bin/g++ \
-B build
```
And it effectively generated the `opencv_python3` target I was looking for. See build output (truncated):
```bash
--     To be built:    [...] ml objdetect optflow photo plot **python3** [...]
--     Disabled:                    cudacodec java_bindings_generator world
--     Disabled by dependency:      -
--     Unavailable:                 cnn_3dobj cvv java js matlab ovis python2 sfm viz
```
